### PR TITLE
Implement TableView.footerView and minor fixes

### PR DESCRIPTION
**Pull request by infosia**
Part of [TIMOB-19962](https://jira.appcelerator.org/browse/TIMOB-19962)

- Add `TableView.footerTitle` and `TableView.footerView`
- Fix: Crash at ListView/TableView selection when footer is visible
- Fix: ListView/TableView shows non-editable checkbox on Store app

```javascript
var win = Ti.UI.createWindow({ backgroundColor: 'green' })
primeSection = Ti.UI.createTableViewSection({
    headerTitle: 'Prime Numbers Header',
    footerTitle: 'Prime Numbers Footer'
}),
fibSection = Ti.UI.createTableViewSection({
   headerView: Ti.UI.createLabel({ text: 'Fibonacci Numbers Header' }),
   footerView: Ti.UI.createLabel({ text: 'Fibonacci Numbers Footer' })
}),
table = Ti.UI.createTableView({ data: [primeSection, fibSection] });

primeSection.add(Ti.UI.createTableViewRow({ title: '2' }));
primeSection.add(Ti.UI.createTableViewRow({ title: '3' }));
primeSection.add(Ti.UI.createTableViewRow({ title: '5' }));
primeSection.add(Ti.UI.createTableViewRow({ title: '7' }));

fibSection.add(Ti.UI.createTableViewRow({ title: '1' }));
fibSection.add(Ti.UI.createTableViewRow({ title: '1' }));
fibSection.add(Ti.UI.createTableViewRow({ title: '2' }));
fibSection.add(Ti.UI.createTableViewRow({ title: '3' }));

table.addEventListener('click', function (e) {
    alert(JSON.stringify(e.rowData.title));
});

win.add(table);
win.open();
```

### CODE REVIEW

File: Source/UI/src/TableView.cpp [hunk 1]
```diff
@@ -226,6 +240,16 @@ namespace TitaniumWindows
 			return header;
 		}
 
+		Controls::ListViewHeaderItem^ TableView::createDefaultSectionFooter(const std::shared_ptr<Titanium::UI::TableViewSection>& section)
+		{
+			Controls::ListViewHeaderItem^ footer = ref new Controls::ListViewHeaderItem();
+			auto footerText = ref new Controls::TextBlock();
+			footerText->Text = Utility::ConvertUTF8String(section->get_footerTitle());
+			footerText->FontSize = 28; // Change this?
```
Comment (sgtcoolguy) [inline Source/UI/src/TableView.cpp]: Not sure what's a recommended value here. I did find this page: https://msdn.microsoft.com/en-us/windows/uwp/controls-and-patterns/fonts

It's hard to know what a footer should be most like in that table. But it does seem to suggest that at larger font (24 epx+) sizes, a light/semilight version of the Segoe UI font should be used.

**Comment by sgtcoolguy** [comment]
For future reference when we come upon font sizes/styles - we should look at whatever guidelines they have. I found this PDF too: https://go.microsoft.com/fwlink/p/?LinkId=760679

It's difficult because it's not like they lay out every possible scenario. But a footer of a ListView seems like maybe a Subtitle to me? That'd mean a 20 EP font size, Segoe UI Regular. But who knows, the "font ramp" they list is really for text blocks more than anything.

**Comment by sgtcoolguy** [comment]
*CR* looks good, @garymathews can you please for an *FR* and merge if good?

**Comment by garymathews** [comment]
FR: **PASS**

**Pull request merged**